src/extra/utils/ftd.pyx

```diff
@@ -0,0 +1,354 @@
+# cython: boundscheck=False, wraparound=False, cdivision=True
+
+from cython cimport floating
+from cython.view cimport contiguous
+from libc.math cimport M_PI, sin, fabs, floor, ceil
+
+import numpy as np
+
+
+"""Edge interpolation method."""
+cpdef enum EdgeInterpolation:
+    NEAREST = 0
+    LINEAR = 1
+    SPLINE = 2
+    SINC = 3
+
+ctypedef floating data_t
+ctypedef floating[::contiguous] data_array_t
+
+
+cdef inline double _sinc(double x) nogil:
+    """Normalized sinc function."""
+
+    if x != 0.0:
+        return sin(x*M_PI) / (x*M_PI)
+    else:
+        return 1.0
+
+
+def sinc_interpolate(
+    double[::contiguous] y_sampled, double[::contiguous] x_interp,
+    double[::contiguous] y_interp = None, int window=100
+):
+    # TODO: Useful function that should go somewhere, but doesn't
+    # actually belong to this module.
+
+    if y_interp is None:
+        y_interp = np.zeros(len(x_interp), dtype=np.float64)
+
+    cdef int i, k, \
+        sampling_start = max(<int>floor(x_interp[0]) - window, 0), \
+        sampling_end = min(<int>ceil(x_interp[0]) + window, y_sampled.shape[0])
+
+    for i in range(y_interp.shape[0]):
+        for k in range(sampling_start, sampling_end):
+            y_interp[i] += y_sampled[k] * _sinc(k - x_interp[i])
+
+    return np.asarray(y_interp)
+
+
+cdef int _cfd_fast_pos(
+    data_array_t signal, data_array_t edges, data_array_t amplitudes,
+    data_t threshold, int delay, int width, data_t fraction, data_t walk,
+    int interp
+) nogil:
```

Comment:
`interp` is unused (same for the matching _neg version below).

Is amplitudes meant to be filled in with something?

Reply:
Yes, `interp` and `amplitudes` is a recent addition to the interface I've only added in the DLE implementation yet. As there's plenty of code that expects both functions to have the same interface, I added stubs there. Started on minimal code for `interp` in fa1f645.